### ISSUE TITLE
Cannot download localhost files on android

### ISSUE THREAD

**Issue by burakmartin** [781flyingdutchman/background_downloader#120]
**Describe the bug**
Cannot download files from "localhost" urls on android

**To Reproduce**

1. flutter pub create .
2. flutter pub add background_downloader
3. open lib/main.dart
4. replace with the following:

```dart
import 'package:background_downloader/background_downloader.dart';
import 'package:flutter/material.dart';

void main() {
  runApp(const MainApp());
}

class MainApp extends StatelessWidget {
  const MainApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: Center(
          child: ElevatedButton(
            child: const Text("Download"),
            onPressed: () async {
              final task = DownloadTask(
                filename: "test.pdf",
                baseDirectory: BaseDirectory.applicationDocuments,
                url: "http://localhost:3000/documents/test.pdf",
              );
              FileDownloader().download(task);
            },
          ),
        ),
      ),
    );
  }
}
```

5. start app
6. hit download

**Expected behavior**
Download starts if file is hosted locally at given url, otherwise fails to download

**Logs**

```
I/BackgroundDownloader( 9246): Enqueuing task with id 2999063836
I/BackgroundDownloader( 9246): Invalid url: http://localhost:3000/documents/test.pdf
```

**Additional context**
In development its totally normal to run backends and frontends locally, so downloading files from these local services should also be possible.

** Workaround **
The workaround I found is doing

```dart
myUrl.replace("localhost", "127.0.0.1")
```

before downloading

**Culprit**
The provided code runs totally fine on IOS, making me believe its an android error. Going further into it, I found

```kotlin
// /background_downloader/android/src/main/kotlin/com/bbflight/background_downloader/BackgroundDownloaderPlugin.kt
suspend fun doEnqueue(...): Boolean {
  ...
  if (!Patterns.WEB_URL.matcher(task.url).matches()) {
      Log.i(TAG, "Invalid url: ${task.url}")
      return false
  }
  ...
}
```

The problem is that WEB_URL.matcher does not match "localhost" and thus prints the above logs.

I have three ideas on how this could possibly be fixed. 
1.  convert localhost urls to "127.0.0.1" inside kotlin so user of the library dont have to. But im not to sure if this will always work since I think localhost might resolve to something else
2. add an additional regex to the if clause for localhost urls
3. dont match at all and let the user enter any url they want

**Comment by 781flyingdutchman** [comment]
Thanks for discovering this and for problem-solving.  Per [this](https://stackoverflow.com/questions/163360/regular-expression-to-match-urls-in-java) I wonder if I should replace it with a general RegEx (as listed there) or use `URL url = URL(stringURL);` and catch `MalformedURLException`.  I can't get to changes for a few days, but if you have time and can experiment with this (especially the URL(stringURL) version is appealing to me) then that would be great!  Also, have you solved for the Android requirement for clearText (HTTP vs HTTPS) requirement?

**Comment by 781flyingdutchman** [comment]
Implemented in V7.9.3
